src/battle_dome.c

```diff
@@ -4410,11 +4410,11 @@ static void DisplayTrainerInfoOnCard(u8 flags, u8 trainerTourneyId)
                 else
                     nature = gSaveBlock2Ptr->frontier.domePlayerPartyData[i].nature;
 
-                if (gNatureStatTable[nature][j] > 0)
+                if (gNaturesInfo[nature].statUp == NUM_STATS + 1)
```

Comment:
I'm not sure how this works without taking j into account. Are you sure this is how you intended it? Same for the three similar occurrences below